llvmlite/ir/builder.py

```diff
@@ -134,6 +134,22 @@ def wrapped(self, val, typ, name=''):
     return wrap
 
 
+def _label_suffix(label, suffix):
+    """Returns (label + suffix) or a truncated version if it's too long.
+    Parameters
+    ----------
+    label : str
+        Label name
+    suffix : str
+        Label suffix
+    """
+    if len(label) > 50:
+        nhead = 25
+        return ''.join([label[:nhead], '..', suffix])
```

Comment:
My only concern with this approach is that it is possible to get to get a short cycle (period 4 if `.endif` is the suffix) of repeating labels:
```
thefunc.endif.endif.endif...endif
thefunc.endif.endif.endif...endif.endif
thefunc.endif.endif.endif...endif.endif.endif
thefunc.endif.endif.endif...endif.endif.endif.endif
thefunc.endif.endif.endif...endif
thefunc.endif.endif.endif...endif.endif
thefunc.endif.endif.endif...endif.endif.endif
[...and so on...]
```
Would that create a problem?

Reply:
Label names are automatically deduplicated via https://github.com/numba/llvmlite/blob/21bce30c67a457ad5eac131bdaca769ddee55afe/llvmlite/ir/values.py#L234-L235

Reply:
OK, then this seems fine.